### DISCUSSION TITLE
Composer update ignores linked php

### DISCUSSION THREAD

**Discussion by offizium-berndstorath** [Homebrew/discussions#5307] [Everyday usage] [answered]
### Output of `brew config`

```shell
ORIGIN: https://github.com/Homebrew/brew
HEAD: 54bea636b41f7953c01a7da5f488f08c1b42ffd1
Last commit: 13 hours ago
Core tap JSON: 22 Apr 11:12 UTC
Core cask tap JSON: 22 Apr 11:12 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.44.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ruby@2.7
```

### Description of issue

I want to use php 8.2. So I installed php@8.2 and linked it.
I also want to use composer. So I installed that.

But everytime there is an update for composer. it downloads php 8.3 as a requirement.
This shouldnt happen. And even if so. The php 8.2 that I linked should stay linked

**Comment by SMillerDev** [comment, marked as answer]
> But everytime there is an update for composer. it downloads php 8.3 as a requirement.
> This shouldnt happen. And even if so. The php 8.2 that I linked should stay linked

Yes it should, it depends on PHP and that means it depends on the latest PHP. I think `brew-php-switcher` is a better way for you to pick a prefered PHP version.
